server/athenian/api/experiments/aggregates/worker.py

```diff
@@ -0,0 +1,569 @@
+
+    # SELECT repository_node_id, repository_full_name, node_id, number
+    # FROM github_pull_requests_compat;
+    # WHERE repository_node_id NOT IN (:repository_node_id)
+    mdb_pr_query = select([
+        PullRequest.repository_node_id,
+        PullRequest.repository_full_name,
+        PullRequest.node_id,
+        PullRequest.number,
+    ])
+    if current_repos:
+        log.info("%d repos found!", len(current_repos))
+
+        # TODO: An alternative could be to save the last fetch,
+        # but then we don't have the insertion date of a repo in metadata.
+        condition = and_(
+            PullRequest.repository_node_id.notin_([r.get(0) for r in current_repos]),
+            PullRequest.repository_full_name.in_(repos),
+        )
+    else:
+        log.info("No repos found")
+        condition = PullRequest.repository_full_name.in_(repos)
+
+    mdb_pr_query = mdb_pr_query.where(condition)
+    log.info("Retrieving new PRs for each new repository from %s...",
+             table_fqdn(mdb, PullRequest))
+    prs_for_new_repos = await mdb.fetch_all(mdb_pr_query)
+    if not prs_for_new_repos:
+        log.info("No new PRs found")
+        return
+
+    log.info("%d PRs found!", len(prs_for_new_repos))
+    log.info("Inserting %d new PRs to %s...", len(prs_for_new_repos),
+             table_fqdn(adb, PullRequestStatus))
+    await adb.execute_many(
+        insert(PullRequestStatus),
+        [PullRequestStatus(**pr).create_defaults().explode(with_primary_keys=True)
+         for pr in prs_for_new_repos],
+    )
+    log.info("%d new PRs inserted!", len(prs_for_new_repos))
+
+
+async def _get_outdated_aggregation_prs(
+        sdb: ParallelDatabase, mdb: ParallelDatabase, adb: ParallelDatabase,
+        freshness_threshold_seconds: int,
+) -> PullRequestsCollection:
+    log.info("Retrieving PRs with outdated aggregations from %s (freshness threshold = %ds)...",
+             table_fqdn(adb, PullRequestStatus), freshness_threshold_seconds)
+    # 1. fetch non-fresh prs from aggregates
+    # SELECT repository_node_id, repository_full_name, node_id, number, last_aggregation_timestamp
+    # FROM pull_requests_status
+    # WHERE EXTRACT(EPOCH FROM (NOW() - last_aggregation_timestamp)) > :threshold
+    unfresh_prs = await adb.fetch_all(
+        select([
+            PullRequestStatus.repository_node_id,
+            PullRequestStatus.repository_full_name,
+            PullRequestStatus.node_id,
+            PullRequestStatus.number,
+            PullRequestStatus.last_aggregation_timestamp,
+        ]).where(
+            extract("epoch", (
+                func.now() - PullRequestStatus.last_aggregation_timestamp),
+            ) > freshness_threshold_seconds,
+        ),
+    )
+
+    if not unfresh_prs:
+        log.info("No non-fresh PRs found")
+        return {}, []
+
+    log.info("%d non-fresh PRs aggregations found!", len(unfresh_prs))
+
+    unfresh_prs_df = pd.DataFrame(unfresh_prs)
```

Comment:
There is a handy `read_sql_query` function in `async_read_sql_query.py` to wrap this :+1: